<launch>
<arg name="use_encoder" value="true" />
<arg name="leg_detection" value="true" />
<arg name="torso_detection" value="true" />
<arg name="sound_detection" value="true" />


<node pkg="cartesian_likelihood_field" type="cartesian_likelihood_field" name="cartesian_likelihood_field" output="screen">
        <param name="use_sim_time" value="true" />
        <param name="CartesianLikelihoodGrid/resolution" value="0.25"/>
        <param name="LikelihoodGrid/grid_angle_min" value="-3.14"/>
        <param name="LikelihoodGrid/grid_angle_max" value="3.14"/>
        <param name="LikelihoodGrid/grid_range_min" value="0.0"/>
        <param name="LikelihoodGrid/grid_range_max" value="10.0"/>
        <param name="LikelihoodGrid/update_rate" value="0.25"/>
        <param name="LikelihoodGrid/human_cell_probability" value="0.95"/>
        <param name="LikelihoodGrid/free_cell_probability" value="0.05"/>
        <param name="LikelihoodGrid/unknown_cell_probability" value="0.2"/>
        <param name="LikelihoodGrid/target_detection_probability" value="0.9"/>
        <param name="LikelihoodGrid/false_positive_probability" value="0.01"/>
        <param name="LikelihoodGrid/sensitivity" value="1"/>
        <param name="LikelihoodGrid/leg_range_stdev" value="0.1"/>
        <param name="LikelihoodGrid/leg_angle_stded" value="2.0"/>
        <param name="LikelihoodGrid/torso_range_stded" value="1.0"/>
        <param name="LikelihoodGrid/torso_angle_stdev" value="2.0"/>
        <param name="LikelihoodGrid/sound_range_stdev" value="1.0"/>
        <param name="LikelihoodGrid/sound_angle_stdev" value="2.0"/>
        <param name="loop_rate" value="10"/>
        <param name="motion_model_enable" value="$(arg use_encoder)"/>
        <param name="leg_detection_enable" value="$(arg leg_detection)"/>
        <param name="torso_detection_enable" value="$(arg torso_detection)"/>
        <param name="sound_detection_enable" value="$(arg sound_detection)"/>
        <param name="fuse_multiply" value="false"/>
    </node>
<group if="$(arg leg_detection)">
    <include file="$(find autonomy_leg_detection)/launch/leg_detection.launch"/>
</group>
<group if="$(arg torso_detection)">
    <include file="$(find autonomy_human)/launch/torso_detection.launch"/>
</group>
</launch>
